### ISSUE TITLE
Make this service open to federation

### ISSUE THREAD

**Issue by eschnou** [appdotnet/api-spec#22]
@daltonc vision is spot on when he writes _" I believe that a number of smaller, interoperable social platforms with a clear, sustainable business models will usurp you. "_ However I don't see anything in the current API ideas that leads to an open, federated, ecosystem of smaller players.

Thanks to federation, users can decide in which silo they want to lock their data (their own if they prefer),  and smaller players (like niche/local social networks) can join the ecosystem, growing the network of users. This does not mean that app.net has to be open source. It can remains a closed source service with a business model behind it.

However, interoperability is not something you build on 'top' but has to be thought in the core. Now is the right time to think about these concepts. Having a look at existing projects (statusnet, diaspora, onesocialweb, friendika,...) and specifications (Ostatus, Activitystreams, Webfinger, Salmon, ...) would help. 

A good starting point is to reach out to the W3C Federated Social Web community: http://www.w3.org/community/fedsocweb/

**Comment by eschnou** [comment]
Note that this could also be an opportunity to design one federation API that works well, instead of the patchworks of specs that are (at least to my experience) difficult to integrate. This doesn't need to be designed by committee, it should just work, allowing interoperability with other providers.

**Comment by matthewp** [comment]
eschnou: I agree with you, but be careful of this: http://xkcd.com/927/

**Comment by mcg** [comment]
Came looking for federation API's, didn't see anything. This would be a requirement to even think about using this.

**Comment by steveklabnik** [comment]
Yep. If app.net really wanted to play nicely with everyone, ostatus would be a great way to do it.

**Comment by orianmarx** [comment]
Here's a list of things we've stated we'll support: http://daltoncaldwell.com/a-response-to-brennan-novak

We're still interested in talking about this stuff but clearly App.net is not currently a federated architecture and won't be for some time if ever. On the other hand, we have strong policies in place against siloing of user data and will be making these stronger soon.
